tensorflow/core/util/mkl_util.h

```diff
@@ -296,14 +337,23 @@ class MklDnnShape {
                                    const memory::desc& md2) const {
     mkldnn_memory_desc_t mdd1 = md1.data;
     mkldnn_memory_desc_t mdd2 = md2.data;
-    const char* d1 = reinterpret_cast<const char*>(&mdd1);
-    const char* d2 = reinterpret_cast<const char*>(&mdd2);
 
-    size_t md_size = sizeof(mdd1);
-    for (size_t i = 0; i < md_size; i++) {
-      if (*d1++ != *d2++) {
-        return false;
-      }
+    assert(mdd1.primitive_kind == mkldnn::primitive::kind::memory);
+    assert(mdd2.primitive_kind == mkldnn::primitive::kind::memory);
+    bool base_equal = true && mdd1.ndims == mdd2.ndims &&
```

Comment:
Nit: We don't need `true`.
```suggestion
    bool base_equal = mdd1.ndims == mdd2.ndims &&
```

Reply:
Done